test/kleros/kleros-liquid-extra-views.js

```
@@ -0,0 +1,262 @@
+      subcourtTree.jurorFee,
+      subcourtTree.jurorsForJump,
+      subcourtTree.timesPerPeriod,
+      subcourtTree.sortitionSumTreeK
+    )
+
+    await pinakion.changeController(klerosLiquid.address)
+    extraViews = await KlerosLiquidExtraViews.new(klerosLiquid.address)
+  })
+
+  it('Should query delayed stakes correctly.', async () => {
+    await klerosLiquid.createSubcourt(
+      subcourtTree.parent,
+      subcourtTree.hiddenVotes,
+      subcourtTree.minStake + 1,
+      subcourtTree.alpha,
+      subcourtTree.jurorFee,
+      subcourtTree.jurorsForJump,
+      subcourtTree.timesPerPeriod,
+      subcourtTree.sortitionSumTreeK
+    )
+
+    const extraData = `0x${(0).toString(16).padStart(64, '0')}${(1)
+      .toString(16)
+      .padStart(64, '0')}`
+    await klerosLiquid.createDispute(2, extraData, {
+      value: await klerosLiquid.arbitrationCost(extraData)
+    })
+    await increaseTime(minStakingTime)
+    await klerosLiquid.passPhase()
+
+    await klerosLiquid.setStake(0, subcourtTree.minStake, {
+      from: governor
+    })
+    await klerosLiquid.setStake(1, subcourtTree.minStake + 1, {
+      from: governor
+    })
+
+    const stake0 = await extraViews.stakeOf(governor, 0)
+    const stake1 = await extraViews.stakeOf(governor, 1)
+    expect(stake0).to.eql(web3.toBigNumber(subcourtTree.minStake))
+    expect(stake1).to.eql(web3.toBigNumber(subcourtTree.minStake + 1))
+
+    const jurorData = await extraViews.getJuror(governor)
+    expect(jurorData[0]).to.deep.equal([
+      web3.toBigNumber(1),
+      web3.toBigNumber(2),
+      web3.toBigNumber(0),
+      web3.toBigNumber(0)
+    ])
+    expect(jurorData[1]).to.eql(web3.toBigNumber(subcourtTree.minStake * 2 + 1))
+    expect(jurorData[2]).to.eql(web3.toBigNumber(0))
+    expect(jurorData[3]).to.deep.equal([
+      stake0,
+      stake1,
+      web3.toBigNumber(0),
+      web3.toBigNumber(0)
+    ])
+  })
+
+  it('Should ignore invalid delayed stakes.', async () => {
```

I don't get the purpose of this test at all.
Transactions leading to invalid stakes would just be reverted, wouldn't they?

---

No, in KlerosLiquid, the only invalid delayed stake that would revert is trying to stake in a non-existing subcourt. All other invalid cases fail silently when delayed stakes are executed https://github.com/kleros/kleros/blob/master/contracts/kleros/KlerosLiquid.sol#L464.

---

The view contract should take this into account.

---

Got it.